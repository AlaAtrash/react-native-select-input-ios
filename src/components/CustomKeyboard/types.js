import PropTypes from 'prop-types'

export default {
  buttonsBackgroundColor: PropTypes.string,
  buttonsBorderColor: PropTypes.string,
  buttonsBorderWidth: PropTypes.number,
  buttonsTextColor: PropTypes.string,
  buttonsTextSize: PropTypes.number,
  cancelKeyText: PropTypes.string,
  onCancelPress: PropTypes.func.isRequired,
  onSubmitPress: PropTypes.func.isRequired,
  submitKeyText: PropTypes.string,
  visible: PropTypes.bool.isRequired,
  children: PropTypes.element
}
