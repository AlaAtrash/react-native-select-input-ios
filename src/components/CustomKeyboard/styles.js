import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor:'rgba(52, 52, 52, 0.8)'
    
    // backgroundColor:'red',
    // marginHorizontal:50
  },
  modal: {
    borderRadius:15,
    padding: 8,
    // alignItems: 'center',
    // justifyContent: 'center',
    backgroundColor: 'white',
    // width:300
    // marginHorizontal:50
  },
  buttonview: {
    padding: 0,
    // borderTopWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: 0,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: '#CCCFD6',
    borderColor: '#CCCFD6',
    // borderTopColor: 'lightgrey',
  }
})

export default styles
