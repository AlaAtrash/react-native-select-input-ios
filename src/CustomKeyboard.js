/**
 * CustomKeybard
 * https://github.com/markuswind/react-native-select-input
 */

import styles from './stylesheets/customKeyboard.css.js';

import PropTypes from 'prop-types';
import React, { Component } from 'react';

import {
  Modal,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View
} from 'react-native';

class CustomKeyboard extends Component {
  constructor(props) {
     super(props);

    this.state = {
      visible: false,
    };
  }

  onCancelPress() {
    this.setVisible(false);
    this.props.onSubmitPress();
  }

  onSubmitPress() {
    this.setVisible(false);
    this.props.onSubmitPress();
  }

  setVisible(visible) {
    this.setState({
      visible: visible
    });
  }

  render() {
    let props = this.props;
    let state = this.state;

    return (
      <Modal animationType={'slide'} transparent={true} visible={state.visible}>
        <TouchableWithoutFeedback onPress={this.onCancelPress.bind(this)}>
          <View style={styles.container}>
            <View style={styles.modal}>
              <View style={[styles.buttonview, { backgroundColor: props.buttonsBackgroundColor }]}>
                <TouchableOpacity onPress={this.onCancelPress.bind(this)}>
                  <Text style={{ color: props.buttonTextColor }}>
                    {props.cancelKeyText}
                  </Text>
                </TouchableOpacity>

                <TouchableOpacity onPress={this.onSubmitPress.bind(this)}>
                  <Text style={{ color: props.buttonTextColor }}>
                    {props.returnKeyText}
                  </Text>
                </TouchableOpacity>
              </View>

              <View>
                  {props.children}
              </View>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </Modal>
    );
  }
}

CustomKeyboard.propTypes =  {
  buttonsBackgroundColor: PropTypes.string,
  buttonTextColor:        PropTypes.string,
  onCancelPress:          PropTypes.func.isRequired,
  onSubmitPress:          PropTypes.func.isRequired,
};

export default CustomKeyboard;
