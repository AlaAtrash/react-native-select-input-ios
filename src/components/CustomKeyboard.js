/**
 * CustomKeyboard
 * https://github.com/markuswind/react-native-select-input
 */

import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Modal, TouchableWithoutFeedback, View } from 'react-native'

import KeyboardButton from './KeyboardButton.js'
import styles from './../stylesheets/customKeyboard.css.js'

class CustomKeyboard extends Component {
  constructor(props) {
    super(props)
  }

  onCancelPress = () => {
    const { onCancelPress } = this.props
    onCancelPress()
  }

  onSubmitPress = () => {
    const { onSubmitPress } = this.props
    onSubmitPress()
  }

  renderKeyBoardButton = (onPress, text, textAlign) => {
    const { buttonsTextColor, buttonsTextSize } = this.props

    return (
      <KeyboardButton
        color={buttonsTextColor}
        onPress={onPress}
        text={text}
        textAlign={textAlign}
        textSize={buttonsTextSize}
      />
    )
  }

  render() {
    const {
      buttonsBackgroundColor,
      buttonsBorderColor,
      buttonsBorderWidth,
      visible,
      cancelKeyText,
      submitKeyText,
      children
    } = this.props

    const buttonsViewStyle = {
      backgroundColor: buttonsBackgroundColor,
      borderColor: buttonsBorderColor,
      borderBottomWidth: buttonsBorderWidth
    }

    return (
      <Modal animationType={'slide'} transparent={true} visible={visible}>
        <TouchableWithoutFeedback onPress={this.onCancelPress.bind(this)}>
          <View style={styles.container}>
            <View style={styles.modal}>
              <View style={[styles.buttonview, buttonsViewStyle]}>
                {this.renderKeyBoardButton(
                  this.onCancelPress,
                  cancelKeyText,
                  'left'
                )}
                {this.renderKeyBoardButton(
                  this.onSubmitPress,
                  submitKeyText,
                  'right'
                )}
              </View>

              <View>{children}</View>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </Modal>
    )
  }
}

CustomKeyboard.propTypes = {
  buttonsBackgroundColor: PropTypes.string,
  buttonsBorderColor: PropTypes.string,
  buttonsBorderWidth: PropTypes.number,
  buttonsTextColor: PropTypes.string,
  buttonsTextSize: PropTypes.number,
  cancelKeyText: PropTypes.string,
  onCancelPress: PropTypes.func.isRequired,
  onSubmitPress: PropTypes.func.isRequired,
  submitKeyText: PropTypes.string,
  visible: PropTypes.bool.isRequired,
  children: PropTypes.element
}

export default CustomKeyboard
