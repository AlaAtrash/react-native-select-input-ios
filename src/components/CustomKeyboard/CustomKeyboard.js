import React, { Component } from 'react'
import { Dimensions, Modal, TouchableWithoutFeedback, View,Text ,useWindowDimensions} from 'react-native'

import KeyboardButton from '../KeyboardButton'

import propTypes from './types.js'
import styles from './styles.js'

class CustomKeyboard extends Component {
  
  state = {
    width: Dimensions.get('window').width,
  
  }
  

  updateDimensions = () => {
    this.setState({
      width: Dimensions.get('window').width
    })
  }

  onCancelPress = () => {
    const { onCancelPress } = this.props
    onCancelPress()
  }

  onSubmitPress = () => {
    const { onSubmitPress } = this.props
    onSubmitPress()
  }

  render() {
    const { width } = this.state
    const {
      buttonsTextStyle,
      buttonsViewStyle,
      visible,
      cancelKeyText,
      submitKeyText,
      children,
      titleModal
    } = this.props

    return (
    
      <Modal
        animationType={'fade'}
        transparent={true}
        visible={visible}
        onOrientationChange={this.updateDimensions}
        supportedOrientations={[
          'portrait',
          'landscape',
          'landscape-left',
          'landscape-right'
        ]}
      >
        <TouchableWithoutFeedback>
          <View style={styles.container}>
            <View style={[styles.modal,{width:width>600?500:"95%"}]}>
              <View style={[styles.buttonview, buttonsViewStyle]}>
                <KeyboardButton
                  onPress={this.onCancelPress}
                  text={cancelKeyText}
                  textAlign={'left'}
                  textStyle={buttonsTextStyle}
                />
              <Text style={{alignSelf:"center",fontSize:18,fontWeight:'600'}}>{titleModal}</Text>
                <KeyboardButton
                  onPress={this.onSubmitPress}
                  text={submitKeyText}
                  textAlign={'right'}
                  textStyle={buttonsTextStyle}
                />
              </View>

              <View>{children}</View>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </Modal>
    
    )
  }
}

CustomKeyboard.propTypes = propTypes

export default CustomKeyboard
