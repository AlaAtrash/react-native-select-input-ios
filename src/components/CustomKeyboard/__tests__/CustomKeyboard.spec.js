import React from 'react'
import renderer from 'react-test-renderer'

import CustomKeyboard from '../CustomKeyboard'

const props = {
  buttonsBackgroundColor: '#AAAAAA',
  buttonsBorderColor: '#BBBBBB',
  buttonsBorderWidth: 1,
  buttonsTextColor: '#CCCCCC',
  buttonsTextSize: 13,
  cancelKeyText: 'cancelKeyText',
  onCancelPress: jest.fn(),
  onSubmitPress: jest.fn(),
  submitKeyText: 'submitKeyText',
  visible: false,
  children: null
}

describe('CustomKeyboard', () => {
  describe('rendering', () => {
    test('renders correctly with given props', () => {
      const customKeyboard = renderer.create(<CustomKeyboard {...props} />)
      expect(customKeyboard).toMatchSnapshot()
    })
  })

  describe('callbacks', () => {
    const onCancelPress = jest.fn()
    const onSubmitPress = jest.fn()

    const customKeyboard = renderer
      .create(
        <CustomKeyboard
          {...props}
          onCancelPress={onCancelPress}
          onSubmitPress={onSubmitPress}
        />
      )
      .getInstance()

    test('call onCancelPress prop correctly', () => {
      customKeyboard.onCancelPress()
      expect(onCancelPress).toBeCalled()
    })

    test('call onSubmitPress prop correctly', () => {
      customKeyboard.onSubmitPress()
      expect(onSubmitPress).toBeCalled()
    })
  })
})
