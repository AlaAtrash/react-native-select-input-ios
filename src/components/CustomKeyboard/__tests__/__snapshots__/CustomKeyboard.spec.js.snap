// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomKeyboard rendering renders correctly with given props 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "justifyContent": "center",
          "padding": 0,
          "width": 750,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderTopColor": "lightgrey",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 0,
              "width": 750,
            },
            Object {
              "backgroundColor": "#AAAAAA",
              "borderBottomWidth": 1,
              "borderColor": "#BBBBBB",
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#CCCCCC",
                "fontSize": 13,
                "padding": 8,
                "textAlign": "left",
                "width": 187.5,
              }
            }
          >
            cancelKeyText
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#CCCCCC",
                "fontSize": 13,
                "padding": 8,
                "textAlign": "right",
                "width": 187.5,
              }
            }
          >
            submitKeyText
          </Text>
        </View>
      </View>
      <View />
    </View>
  </View>
</Modal>
`;
